### PR TITLE
Do not apply scroll offset corrections with zero values during sliver list layout

### LINKED ISSUE #10547: ListView content vanishes when scrolled up on Android
## Steps to Reproduce
Create a `ListView` widget. 
**Important!!** Wrap each child in `AspectRatio`.
**Important!!** Make enough children to have a content at least in 3 times bigger then a screen 

I created a sample project: https://github.com/nonameden/flutter_listview_issue
Video with sample: https://youtu.be/Ux2zg7XKAOY

Issue only happens on Android. On iOS all work as expected.

### flutter analyze

> Analyzing /Users/nonameden/Documents/AndroidProjects/listview_issue...
> No issues found!
> Ran in 4.4s

## Flutter Doctor

> [✓] Flutter (on Mac OS X 10.12.4 16E195, channel alpha)
>     • Flutter at /Users/nonameden/Documents/flutter
>     • Framework revision 973304d5fb (3 weeks ago), 2017-06-02 10:43:54 -0700
>     • Engine revision 1f2aa07571
>     • Tools Dart version 1.23.0-dev.11.11
> 
> [✓] Android toolchain - develop for Android devices (Android SDK 25.0.3)
>     • Android SDK at /Users/nonameden/Library/Android/sdk
>     • Platform android-25, build-tools 25.0.3
>     • ANDROID_HOME = /Users/nonameden/Library/Android/sdk
>     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version: OpenJDK Runtime Environment (build 1.8.0_112-release-b06)
> 
> [✓] iOS toolchain - develop for iOS devices (Xcode 8.2.1)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Xcode 8.2.1, Build version 8C1002
>     • ios-deploy 1.9.1
>     • CocoaPods version 1.2.1
> 
> [✓] Android Studio (version 2.3)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Gradle version 3.2
>     • Java version: OpenJDK Runtime Environment (build 1.8.0_112-release-b06)
> 
> [✓] IntelliJ IDEA Ultimate Edition (version 2017.1.3)
>     • Dart plugin version 171.4424.63
>     • Flutter plugin version 13.1
> 
> [✓] Connected devices
>     • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/10547

### CODE REVIEW

**Comment by jason-simmons** [comment]
@Hixie

**Comment by tvolkert** [comment]
Looks like this may fix #9665 too (since that's likely a dup of #10547)

File: packages/flutter/lib/src/rendering/sliver.dart [hunk 1]
```diff
@@ -648,7 +648,7 @@ class SliverGeometry {
         buffer.write('hitTestExtent: ${hitTestExtent.toStringAsFixed(1)}, ');
       if (hasVisualOverflow)
         buffer.write('hasVisualOverflow: true, ');
-      buffer.write('scrollOffsetCorrection: ${scrollOffsetCorrection.toStringAsFixed(1)}');
+      buffer.write('scrollOffsetCorrection: ${scrollOffsetCorrection?.toStringAsFixed(1)}');
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver.dart]: Let's just omit the scrollOffsetCorrection unless it's non-null.

**Comment by Hixie** [comment]
This needs a test.

Other than that, LGTM!

**Comment by jason-simmons** [comment]
Added a test

**Comment by Hixie** [comment]
LGTM

**Pull request merged**